modules/skybrowser/src/screenspaceskybrowser.cpp

```diff
@@ -0,0 +1,276 @@
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+#include <modules/skybrowser/include/screenspaceskybrowser.h>
+
+#include <modules/skybrowser/skybrowsermodule.h>
+#include <modules/skybrowser/include/utility.h>
+#include <openspace/engine/globals.h>
+#include <openspace/engine/windowdelegate.h>
+#include <openspace/engine/moduleengine.h>
+#include <openspace/rendering/renderengine.h>
+#include <ghoul/logging/logmanager.h>
+#include <ghoul/misc/dictionaryjsonformatter.h> 
+#include <ghoul/opengl/texture.h>
+#include <optional>
+#include <glm/gtx/color_space.hpp> 
+#include <random>
+
+namespace {
+    constexpr const char* _loggerCat = "ScreenSpaceSkyBrowser";
+
+    constexpr const openspace::properties::Property::PropertyInfo AnimationSpeedInfo =
+    {
+        "AnimationSpeed",
+        "Field Of View Animation Speed",
+        "The factor which is multiplied with the animation of the field of view."
+    };
+    constexpr const openspace::properties::Property::PropertyInfo TextureQualityInfo =
+    {
+        "TextureQuality",
+        "Quality of Texture",
+        "A parameter to set the resolution of the texture. 1 is full resolution and "
+        "slower framerate. Lower value means lower resolution of texture and faster "
+        "frame rate."
+    };
+
+    struct [[codegen::Dictionary(ScreenSpaceSkyBrowser)]] Parameters {
+        // [[codegen::verbatim(AnimationSpeedInfo.description)]]
+        std::optional<double> animationSpeed;
+
+        // [[codegen::verbatim(TextureQualityInfo.description)]]
+        std::optional<float> textureQuality;
+    };
+
+#include "screenspaceskybrowser_codegen.cpp"
+} // namespace
+
+glm::ivec3 randomBorderColor(glm::ivec3 highlight) {
+    // Generate a random border color with sufficient lightness and a n
+    std::random_device rd;
+    // Hue is in the unit degrees [0, 360]
+    std::uniform_real_distribution<float> hue(0.f, 360.f);
+   
+    // Value in saturation are in the unit percent [0,1]
+    float value = 0.9f; // Brightness
```

Comment:
maybe rename this variable to `brightness` instead? Then the comment isn't needed? :)